syntax = "proto3";

message Ship{
    bool isInPort = false;
    uint32 remainingContainersNo = 0;
}

message StorageYard{
    uint32 containersNo = 0;
}

message LoadingPoint{
    enum LoadingPoint_ID{
        SHIP_1 = 0;
        SHIP_2 = 1;
        SHIP_3 = 2;
        STORAGE_YARD_1 = 3;
        STORAGE_YARD_1 = 4;
        TRANSIT_POINT_AFRICA = 5;
        TRANSIT_POINT_EUROPA = 6;
        TRANSIT_POINT_ASIA = 7;
        TRANSIT_POINT_AMERICA = 8;
    }

    LoadingPoint_ID ID = 0;
    bool busy = false;
}

message TransitPoint{
    enum Port_ID {
        AFRICA = 0;
        EUROPA = 1;
        ASIA = 2;
        AMERICA = 3;
    }

    Port_ID ID = AFRICA;
    uint32 containersNo;
}

message PortState{
    Ship ship;
    StorageYard storageYard;
    repeated TransitPoint = 4;
    repeated LoadingPoint = 9;
}